src/Networking/NexusMods.Networking.NexusWebApi/RunUpdateCheck.cs

```diff
@@ -49,28 +49,56 @@ public static async Task<PerFeedCacheUpdaterResult<PageMetadataMixin>> CheckForM
     public static async Task UpdateModFilesForOutdatedPages(IDb db, ITransaction tx, ILogger logger, INexusGraphQLClient gqlClient, PerFeedCacheUpdaterResult<PageMetadataMixin> result, CancellationToken cancellationToken)
     {
         // Note(sewer): Undetermined items may be removed items from the site; or
-        // caused by programmer error, so wr should log these whenever possible,
+        // caused by programmer error, so we should log these whenever possible,
         // but they should not cause a critical error; in case it's simply the result
         // of mod removal such as DMCA takedown.
-        foreach (var mixin in result.UndeterminedItems)
+        using var semaphore = new SemaphoreSlim(16);
```

Comment:
Why 16? Should we use something like `Environment.ProcessorCount` instead? They are going to execute on the same thread, so it doesn't quite track, but 16 seems arbitrary.

Reply:
[This is for others reading]
This is an I/O bound workload, not a CPU bound one.

What we're actually doing is throttling the amount of concurrent requests to the Nexus API, more strictly saying, we only want 16 requests to the API at any given time.

What actually happens is when we await, it yields, and one of the other tasks gets picked up; we're actually limiting the requests to 16 at a time in transit.

The number is arbitrary (as we DM'd to each other too); I'll make a note of this in the code.